File: vijaymamra/vijaymamra/custom_script/delivery_note/delivery_note.js
/ Copyright (c) 2017, Frappe Technologies Pvt. Ltd. and Contributors

// License: GNU General Public License v3. See license.txt


frappe.ui.form.on("Delivery Note Item", {

   bag : function(frm, cdt, cdn) {

		console.log("===========================")

		var child = locals[cdt][cdn];

		if(child){

			if (child.bag){

				frappe.call({

						method:"vijaymamra/custom_script/delivery_note/delivery_note"
						args:{

							"bag_qty": child.bag,

							"weight_per_bag": child.weight_per_bag,

							
						},

						callback: function(r) {

							if(r.message) {

								console.log(r.message)

								frappe.model.set_value(cdt, cdn, "qty",r.message);

							}

						}

					})

			} 

		}

	}


});

File: vijaymamra/vijaymamra/custom_script/delivery_note/delivery_note.py

from __future__ import unicode_literals

import frappe

from frappe.model.document import Document

import decimal


@frappe.whitelist()

def calculate_row_quantity(bag_qty, weight_per_bag):

	bag = float(bag_qty)

	weight_per_bag = float(weight_per_bag)

	row_qty = bag * weight_per_bag

	return round(row_qty)

